packages/ton-client/src/shaders/render.fs

```diff
@@ -1,80 +1,84 @@
 #version 300 es
 
-precision highp float; 
+precision highp float;
 
 uniform vec3 camPosition;
 uniform float time;
 
-uniform vec2 swipeA; 
-uniform vec2 swipeB; 
+uniform vec2 swipeA;
+uniform vec2 swipeB;
 
-in vec3 ray; 
+uniform vec3 touchManipulationState;
 
-out vec4 rgba; 
+in vec3 ray;
+
+out vec4 rgba;
 
 // function for rotating around y axis
-float sdf(vec3); 
+float sdf(vec3, vec3);
```

Comment:
uniforms are global. 
We #include the sdf function. This is similar to C++ simply inserting that file in this shader here. 
So, you don't have to pass uniforms around. 
You can directly access them in the sdf file and function and you should do so in my opinion

Reply:
we could also try to define uniforms at the top of the sdf file. 
this may require us to move the include up above the main function. 
this may mess with line numbers on glsl errors (which makes debugging shaders tedious) - I'd like to try whether this is the case

Reply:
will try it tomorrow